contracts/poap/src/msg.rs

```diff
@@ -0,0 +1,401 @@
+    /// If the mint is enabled, allow the user to mint the poap by themself.
+    /// It's disabled before the start of the event and after the event's end.
+    Mint {},
+    /// Allows the contract's admin or the minter to mint a POAP for a specific recipient.
+    /// It's disabled before the start of the event and after the event's end.
+    MintTo { recipient: String },
+    /// Message that allows the event's creator to change the time frame of the event
+    /// if it's not started or finished.
+    UpdateEventInfo {
+        start_time: Timestamp,
+        end_time: Timestamp,
+    },
+    /// Allows the contract's admin to transfer the admin rights to another user.
+    UpdateAdmin { new_admin: String },
+    /// Allows the contract's admin to transfer the minting rights to another user.
+    UpdateMinter { new_minter: String },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    /// Returns the configuration info as a [`QueryConfigResponse`].
+    Config {},
+    /// Returns the event info as a [`QueryEventInfoResponse`].
+    EventInfo {},
+    /// Returns the amount of poaps minted from `user` as [`QueryMintedAmountResponse`].
+    MintedAmount { user: String },
+}
+
+/// Response to [`QueryMsg::Config`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct QueryConfigResponse {
+    /// Address of the contract administrator.
+    pub admin: Addr,
+    /// Address of the entity that is allowed to use [`ExecuteMsg::MintTo`].
+    pub minter: Addr,
+    /// Tells if the users can execute the [`ExecuteMsg::Mint`].
+    pub mint_enabled: bool,
+    /// The maximus number of poap that an user can request.
+    pub per_address_limit: u32,
+    /// Id of the cw721 contract that this contract has initialized.
+    pub cw721_contract_code: Uint64,
+    /// Address of the cw721 contract that this contract is using to
+    /// mint the poaps.
+    pub cw721_contract: Addr,
+}
+
+/// Response to [`QueryMsg::EventInfo`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct QueryEventInfoResponse {
+    /// Address of who created the event.
+    pub creator: Addr,
+    /// Time at which the event starts.
+    pub start_time: Timestamp,
+    /// Time at which the event ends.
+    pub end_time: Timestamp,
+    /// IPFS uri where the event's metadata are stored
+    pub event_uri: String,
+}
+
+/// Response to [`QueryMsg::MintedAmount`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct QueryMintedAmountResponse {
+    /// Address for which the request was made.
+    pub user: Addr,
+    /// Amount of poaps minted from the user.
+    pub amount: u32,
+}
+
+impl InstantiateMsg {
+    /// Checks that the data inside the message are coherent.
+    /// NOTE: This function don't checks if the address are valid.
+    pub fn validate(&self) -> Result<(), ContractError> {
```

Comment:
cw721 code id should be checked not to be 0 here as well.

Reply:
I think is not needed to add an extra error code for that, if an user provides an invalid contract code ther will be the wasm instantiate message that will fail. What you think?